driver/reducecalculation_driver.hpp

```diff
@@ -189,11 +228,17 @@ int ReduceCalculationDriver<Tgpu, Tref>::GetandSetData()
     reduceCalculationOp =
         static_cast<miopenReduceCalculationOp_t>(inflags.GetValueInt("ReduceCalculationOp"));
 
+    if(isLogicalCalculation && !(reduceCalculationOp == MIOPEN_REDUCE_CALCULATION_ANY))
+    {
+        MIOPEN_THROW(
+            "Mismatch type_calculation and reduce_calculation_op: Logical calculation only "
+            "supports reduce calculation operation type MIOPEN_REDUCE_CALCULATION_ANY.");
```

Comment:
```suggestion
        MIOPEN_THROW(miopenbadparam,
            "Mismatch type_calculation and reduce_calculation_op: Logical calculation only "
            "supports reduce calculation operation type MIOPEN_REDUCE_CALCULATION_ANY.");
```

Reply:
pls check my new implementation: isLogicalCalculation is determined based on `OpType`, which is passed through flag. Therefore, I dont need to handle to throw this mismatch type exception anymore